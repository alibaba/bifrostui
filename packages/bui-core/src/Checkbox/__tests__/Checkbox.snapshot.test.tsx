import { snapshotTest } from 'testing';

describe('Checkbox snapshot', () => {
  snapshotTest('Checkbox');
});
