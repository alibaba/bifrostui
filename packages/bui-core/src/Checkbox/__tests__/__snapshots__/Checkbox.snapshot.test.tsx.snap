// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox snapshot Checkbox demo snapshot 0 1`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    className="bui-checkbox bui-checkbox-label-right"
  >
    <input
      checked={false}
      className="bui-checkbox-input"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="bui-checkbox-icon"
    >
      <svg
        className="bui-svg-icon"
        color="#ced1d6"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-checkbox-label"
    >
      Checkbox
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 2`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    className="bui-checkbox bui-checkbox-label-right"
  >
    <input
      checked={false}
      className="bui-checkbox-input my-input"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="bui-checkbox-icon"
    >
      <svg
        className="bui-svg-icon"
        color="#ced1d6"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-checkbox-label"
    >
      Checkbox
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 3`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    className="bui-checkbox bui-checkbox-label-right"
  >
    <input
      checked={false}
      className="bui-checkbox-input"
      name="myCheckbox"
      onChange={[Function]}
      type="checkbox"
      value="tpp"
    />
    <div
      className="bui-checkbox-icon"
    >
      <svg
        className="bui-svg-icon"
        color="#ced1d6"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-checkbox-label"
    >
      Checkbox
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 4`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <div
      className="bui-checkbox bui-checkbox-label-top"
    >
      <input
        checked={false}
        className="bui-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ced1d6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        文案在上边
      </div>
    </div>
    <div
      className="bui-checkbox bui-checkbox-label-right"
    >
      <input
        checked={false}
        className="bui-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ced1d6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        文案在右边
      </div>
    </div>
    <div
      className="bui-checkbox bui-checkbox-label-bottom"
    >
      <input
        checked={false}
        className="bui-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ced1d6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        文案在下边
      </div>
    </div>
    <div
      className="bui-checkbox bui-checkbox-label-left"
    >
      <input
        checked={false}
        className="bui-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ced1d6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        文案在左边
      </div>
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 5`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    className="bui-checkbox bui-checkbox-label-right"
  >
    <input
      checked={false}
      className="bui-checkbox-input"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="bui-checkbox-icon"
    >
      <svg
        className="bui-svg-icon"
        color="#ccc"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M62.6 11C78.1 11 88 23.4 88 40.1c0 13-11.7 27.2-34.8 43.3-3.1 2.2-7.2 2.2-10.3 0C19.7 67.3 8 53.1 8 40.1 8 23.4 17.9 11 33.4 11c5.6 0 9.4 2 14.6 6.4C53.2 13 57 11 62.6 11zm0 5.9c-3.9 0-6.6 1.4-11 5.2-.2.2-1.4 1.2-1.7 1.4-1.1 1-2.8 1-3.9 0-.3-.3-1.5-1.3-1.7-1.4-4.4-3.7-7.1-5.2-11-5.2C21.6 16.9 14 26.4 14 40c0 10.4 10.6 23.3 32.3 38.3 1 .7 2.4.7 3.4 0C71.4 63.3 82 50.4 82 40c0-13.6-7.6-23.1-19.4-23.1z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-checkbox-label"
    >
      Checkbox
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 6`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    className="bui-checkbox bui-checkbox-label-right bui-checkbox-disabled"
  >
    <input
      checked={false}
      className="bui-checkbox-input"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="bui-checkbox-icon"
    >
      <svg
        className="bui-svg-icon"
        color="#ced1d6"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-checkbox-label"
    >
      Checkbox
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 7`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    className="bui-checkbox-group"
  >
    <div
      className="bui-checkbox bui-checkbox-label-right"
    >
      <input
        checked={false}
        className="bui-checkbox-input"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
        value="水门桥"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ced1d6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        水门桥
      </div>
    </div>
    <div
      className="bui-checkbox bui-checkbox-label-right"
    >
      <input
        checked={false}
        className="bui-checkbox-input"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
        value="淘票票"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ced1d6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        淘票票
      </div>
    </div>
    <div
      className="bui-checkbox bui-checkbox-label-right"
    >
      <input
        checked={false}
        className="bui-checkbox-input"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
        value="喜洋洋"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ced1d6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 88C25.9 88 8 70.1 8 48S25.9 8 48 8s40 17.9 40 40-17.9 40-40 40zm0-6c18.8 0 34-15.2 34-34S66.8 14 48 14 14 29.2 14 48s15.2 34 34 34z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        喜洋洋
      </div>
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 8`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <button
      className="bui-btn bui-btn-outlined bui-btn-medium bui-btn-default"
      onClick={[Function]}
      type="button"
    >
      Uncheck
       All
    </button>
    <div
      className="bui-checkbox-group"
    >
      <div
        className="bui-checkbox bui-checkbox-label-right"
      >
        <input
          checked={true}
          className="bui-checkbox-input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="水门桥"
        />
        <div
          className="bui-checkbox-icon"
        >
          <svg
            className="bui-svg-icon"
            color="#ff335c"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
              }
            }
            focusable="false"
            viewBox="0 0 96 96"
          />
        </div>
        <div
          className="bui-checkbox-label"
        >
          水门桥
        </div>
      </div>
      <div
        className="bui-checkbox bui-checkbox-label-right"
      >
        <input
          checked={true}
          className="bui-checkbox-input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="淘票票"
        />
        <div
          className="bui-checkbox-icon"
        >
          <svg
            className="bui-svg-icon"
            color="#ff335c"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
              }
            }
            focusable="false"
            viewBox="0 0 96 96"
          />
        </div>
        <div
          className="bui-checkbox-label"
        >
          淘票票
        </div>
      </div>
      <div
        className="bui-checkbox bui-checkbox-label-right"
      >
        <input
          checked={true}
          className="bui-checkbox-input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="喜洋洋"
        />
        <div
          className="bui-checkbox-icon"
        >
          <svg
            className="bui-svg-icon"
            color="#ff335c"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
              }
            }
            focusable="false"
            viewBox="0 0 96 96"
          />
        </div>
        <div
          className="bui-checkbox-label"
        >
          喜洋洋
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <button
      className="bui-btn bui-btn-outlined bui-btn-medium bui-btn-default"
      onClick={[Function]}
      type="button"
    >
      取消
    </button>
    <div
      className="bui-checkbox bui-checkbox-label-right"
    >
      <input
        checked={true}
        className="bui-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ff335c"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        淘票票
      </div>
    </div>
  </div>
</div>
`;

exports[`Checkbox snapshot Checkbox demo snapshot 0 9`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
    }
  }
>
  <div
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <div>
      CheckboxGroup：
    </div>
    <div
      className="bui-checkbox-group"
    >
      <div
        className="bui-checkbox bui-checkbox-label-right"
      >
        <input
          checked={true}
          className="bui-checkbox-input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="水门桥"
        />
        <div
          className="bui-checkbox-icon"
        >
          <svg
            className="bui-svg-icon"
            color="#ff335c"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
              }
            }
            focusable="false"
            viewBox="0 0 96 96"
          />
        </div>
        <div
          className="bui-checkbox-label"
        >
          水门桥
        </div>
      </div>
      <div
        className="bui-checkbox bui-checkbox-label-right"
      >
        <input
          checked={true}
          className="bui-checkbox-input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="淘票票"
        />
        <div
          className="bui-checkbox-icon"
        >
          <svg
            className="bui-svg-icon"
            color="#ff335c"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
              }
            }
            focusable="false"
            viewBox="0 0 96 96"
          />
        </div>
        <div
          className="bui-checkbox-label"
        >
          淘票票
        </div>
      </div>
      <div
        className="bui-checkbox bui-checkbox-label-right"
      >
        <input
          checked={true}
          className="bui-checkbox-input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="喜洋洋"
        />
        <div
          className="bui-checkbox-icon"
        >
          <svg
            className="bui-svg-icon"
            color="#ff335c"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
              }
            }
            focusable="false"
            viewBox="0 0 96 96"
          />
        </div>
        <div
          className="bui-checkbox-label"
        >
          喜洋洋
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      Checkbox：
    </div>
    <div
      className="bui-checkbox bui-checkbox-label-right"
    >
      <input
        checked={true}
        className="bui-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="bui-checkbox-icon"
      >
        <svg
          className="bui-svg-icon"
          color="#ff335c"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M48 8c22.1 0 40 17.9 40 40S70.1 88 48 88 8 70.1 8 48 25.9 8 48 8zm17.1 27.9c-1.2-1.2-3.1-1.2-4.2 0L43 53.8l-7.9-7.9c-1.2-1.2-3.1-1.2-4.2 0s-1.2 3.1 0 4.2l10 10c1.2 1.2 3.1 1.2 4.2 0l20-20c1.2-1.2 1.2-3.1 0-4.2z\\"/>",
            }
          }
          focusable="false"
          viewBox="0 0 96 96"
        />
      </div>
      <div
        className="bui-checkbox-label"
      >
        淘票票
      </div>
    </div>
  </div>
</div>
`;
